OutOfSchool/OutOfSchool.BusinessLogic/Services/ApplicationService.cs

```diff
@@ -979,7 +979,8 @@ await AllowedNewApplicationByChildStatus(applicationDto.WorkshopId, applicationD
                 && Enum.TryParse(additionalData[StatusTitle], out ApplicationStatus applicationStatus))
             {
                 if (applicationStatus == ApplicationStatus.Approved
-                    || applicationStatus == ApplicationStatus.Rejected)
+                    || applicationStatus == ApplicationStatus.Rejected
+                    || applicationStatus == ApplicationStatus.AcceptedForSelection)
```

Comment:
(Design) Please consider if this check could be extracted and reused as a helper method. Currently there are 2 other places where kind of same set of condition is checked - in `ExecuteUpdateAsync()` the `if` wrapping call to `SendApplicationUpdateStatusEmail()`, and in mentioned `SendApplicationUpdateStatusEmail()` the `switch` throwing "unsupported" `ArgumentException` when condition is not met. Here you've introduced the 3rd place. 
If it's not the coincidence and it's expected that logic for all mentioned places should be the same - please make it technically the same and implemented in 1 place.

Reply:
Done